src/vs/workbench/parts/quickopen/browser/gotoLineHandler.ts

```diff
@@ -184,7 +192,11 @@ export class GotoLineHandler extends QuickOpenHandler {
 			this.lastKnownEditorViewState = (<IEditor>editor.getControl()).saveViewState();
 		}
 
-		return TPromise.as(new QuickOpenModel([new GotoLineEntry(searchValue, this.editorService, this)]));
+		const entry = this.instantiationService.createInstance(
```

Comment:
@wprater you do not need to pass in the services, just do: `this.instantiationService.createInstance(GotoLineEntry, searchValue)`

All services will be passed in automatically.